inc/hclib-task.h

```diff
@@ -32,32 +32,20 @@ typedef struct _hclib_task_t {
      * and locality flexible asyncAny
      */
     int is_async_any_type;
```

Comment:
> I'm trying to get this merged in now that I've realized each of Nick's diffs includes all of the preceding diffs.

The changes build off of each other. As long as we go through the pull requests in order, each should be a manageable set of just a few commits. I thought that was clear from the last line in each of my pull requests' comments. The changes are all intertwined (later changes to promises affect the async_future function signatures, which were also updated in the earlier patches). Processing these pull requests in order (vs trying to split them into independent changes) should save us from some very hairy merges later.